tests/wallet/rpc/test_wallet_rpc.py

```diff
@@ -896,7 +896,63 @@ def only_ids(trades):
                 "yc449jqxg4mfgx0sw3p9"
             },
         )
-    ###
+
+
+@pytest.mark.asyncio
+async def test_check_ids(wallet_rpc_environment: WalletRpcTestEnvironment):
+    env: WalletRpcTestEnvironment = wallet_rpc_environment
+    wallet_1_node: WalletNode = env.wallet_1.node
+    wallet_1_rpc: WalletRpcClient = env.wallet_1.rpc_client
+    full_node_api: FullNodeSimulator = env.full_node.api
+    await generate_funds(env.full_node.api, env.wallet_1, 5)
+
+    # DID
+    res = await wallet_1_rpc.create_new_did_wallet(amount=1, name="Profile 1")
+    assert res["success"]
+    did_id_0 = res["my_did"]
+    res = await wallet_1_rpc.check_ids([did_id_0], "DID")
+    assert len(res) == 1
+    assert res[0] == decode_puzzle_hash(did_id_0).hex()
+
+    await farm_transaction_block(full_node_api, wallet_1_node)
+    await full_node_api.wait_for_wallet_synced(wallet_node=wallet_1_node, timeout=15)
+    # NFT
+    res = await wallet_1_rpc.create_new_nft_wallet(None)
+    nft_wallet_id = res["wallet_id"]
+    res = await wallet_1_rpc.mint_nft(
+        nft_wallet_id,
+        None,
+        None,
+        "0xD4584AD463139FA8C0D9F68F4B59F185",
+        ["https://www.chia.net/img/branding/chia-logo.svg"],
+    )
+    assert res["success"]
+
+    spend_bundle = SpendBundle.from_json_dict(json_dict=res["spend_bundle"])
+
+    await farm_transaction(full_node_api, wallet_1_node, spend_bundle)
+
+    await full_node_api.wait_for_wallet_synced(wallet_node=wallet_1_node, timeout=15)
+
+    nft_wallet: WalletProtocol = wallet_1_node.wallet_state_manager.wallets[nft_wallet_id]
+    assert isinstance(nft_wallet, NFTWallet)
+
+    async def have_nfts():
+        return await nft_wallet.get_nft_count() > 0
+
+    await time_out_assert(15, have_nfts, True)
+
+    # Test with the hex version of nft_id
+    nft_id = (await nft_wallet.get_current_nfts())[0].nft_id.hex()
+    res = await wallet_1_rpc.check_ids([nft_id], "NFT")
+    assert len(res) == 1
+    assert res[0] == nft_id
+
+    # COIN
+    coin_id = (await nft_wallet.get_current_nfts())[0].coin.name().hex()
+    res = await wallet_1_rpc.check_ids([coin_id], "COIN")
+    assert len(res) == 1
+    assert res[0] == coin_id
```

Comment:
add a test to check if it doesn't return coins that it doesn't own but they should exist

Reply:
I thought only the coins we owned will be saved in the table. What is the reason for saving the coin we don't own?